packages/vue/src/state/tags/index.js

```diff
@@ -0,0 +1,40 @@
+import { TAG } from '../mutationTypes';
+
+const initialState = {
+  tags: [],
+  error: null,
+  isLoading: false
+};
+
+const state = { ...initialState };
+
+const mutations = {
+  [TAG.LOAD_POPULAR_TAGS_REQUEST](state) {
+    state.isLoading = true;
+  },
+  [TAG.LOAD_POPULAR_TAGS_SUCCESS](state, tags) {
+    state.isLoading = false;
+    state.tags = tags;
+  },
+  [TAG.LOAD_POPULAR_TAGS_ERROR](state, error) {
+    state.isLoading = false;
+    state.error = error;
+  }
+};
+
+const actions = {
+  [TAG.LOAD_POPULAR_TAGS]({ commit }) {
+    commit(TAG.LOAD_POPULAR_TAGS_REQUEST);
+
+    this.app.getPopularTags({
```

Comment:
Hi! I just stumbled upon this great repo and this PR, and I'm heavily interested in creating a proof of concept of this based on Vue. Now, I have this question: did you manage to achieve autocompleition and type hinting after injecting dependencies using a Vuex plugin and accessing it with `this.app`?

Thank you for your time! The whole repo has been really enlightening to me. Appreciated.